pkgs/tools/misc/coreutils/default.nix

```diff
@@ -53,7 +53,7 @@ stdenv.mkDerivation rec {
       ++ optional aclSupport acl.crossDrv
       ++ optional attrSupport attr.crossDrv
       ++ optionals selinuxSupport [ libselinux.crossDrv libsepol.crossDrv ]
-      ++ optional (stdenv.ccCross.libc ? libiconv)
+      ++ optional (stdenv.ccCross or {} ? libc.libiconv)
```

Comment:
I think the Nix `or` operator doesn't help you when an attribute is set to null, unlike the "or" operations in a lot of other languages.  The `or` is just used to specify a backup value in case the attribute path you specified cannot be evaluated because one of the attributes is missing.  Anyway, I'm sure you are able to play around with it and find a simple expression that works.